### PR TITLE
tests script for board

### PR DESCRIPTION
**Pull request by potsrevennil**
[//]: # (SPDX-License-Identifier: CC-BY-4.0)
[//]: # (TODO Customize PR template)

[//]: # (See https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository )

Mainly added a python script for running tests on board. Merged the original tests script into it as well.
Currently it's slow to run the nistkat test on board, due to the bottleneck of IO. Checking the checksum on board would be better.

```
☺  tests --help                                                                              
Usage: tests [OPTIONS] COMMAND [ARGS]...

Options:
  --help  Show this message and exit.

Commands:
  func     Run functional tests
  nistkat  Run nistkat tests
  run      Run for the specified platform and hex file without parsing the
           output
  speed    Run speed tests
  stack    Run stack tests
```
```
☺  tests func --help
Usage: tests func [OPTIONS]

Options:
  -cfg, --platform-cfg PATH  Configuration file of the specified platform
                             [default: hal/stm32f4discovery.cfg]
  -v, --verbose              Show verbose output or not
  -e, --emulate              Emulate on the QEMU or not
  -t, --ntests INTEGER       Number of tests  [default: 1]
  --help                     Show this message and exit.
```

- add marker for tests when running on board
- add python script for running tests on board
- use the python tests script in ci instead
- clean up the original ci test script

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)

### CODE REVIEW

File: test/speed.c [hunk 1]
```diff
@@ -6,6 +6,14 @@
...
 #include <string.h>
 
+#if !defined(MPS2_AN386)
+#define SERIAL_MARKER() {\
+        hal_send_str("$");\
+    }
+#else
+#define SERIAL_MARKER()
+#endif
+
```
Comment (mkannwischer) [inline test/speed.c:16]: maybe move this to the hal.h?

File: scripts/tests [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            lambda output: count(output, "OK"),
+            verbose,
+        )
+    except asyncio.CancelledError:
+        pass
+
+
+@click.command(short_help="Run speed tests", context_settings={"show_default": True})
+@add_options(_shared_options)
+@click.option("-t", "--ntests", default=1, type=int, help="Number of tests")
```
Comment (mkannwischer) [inline scripts/tests:211]: Can you please rename this one to `-i` `--iterations` to be consistent with pqm4.

**Review by mkannwischer — CHANGES_REQUESTED** [review]
Nice work, thank you! 
I guess we want to make this the preferred way to perform tests and benchmarks. Can you please adjust the main README accordingly?


```
$ tests speed -t 100 -v
Traceback (most recent call last):
  File "/tmp/mlkem-c-embedded/scripts/tests", line 290, in <module>
    cli()
  File "/nix/store/r725a3hx83v9k6222xc6kamf8rmzzlxs-python3.11-click-8.1.7/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/r725a3hx83v9k6222xc6kamf8rmzzlxs-python3.11-click-8.1.7/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/nix/store/r725a3hx83v9k6222xc6kamf8rmzzlxs-python3.11-click-8.1.7/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/r725a3hx83v9k6222xc6kamf8rmzzlxs-python3.11-click-8.1.7/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/r725a3hx83v9k6222xc6kamf8rmzzlxs-python3.11-click-8.1.7/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/mlkem-c-embedded/scripts/tests", line 214, in speed
    base_test(
  File "/tmp/mlkem-c-embedded/scripts/tests", line 126, in base_test
    check(file, expect, actual)
  File "/tmp/mlkem-c-embedded/scripts/tests", line 107, in check
    sys.exit(1)
    ^^^
NameError: name 'sys' is not defined
```

File: test/speed.c (hunk 1 above)
Comment (potsrevennil) [reply test/speed.c]: good idea, originally thought that I should add a common header for tests (but that is ugly...)

**Comment by potsrevennil** [comment]
Updated the README, but not sure if it's clear enough...

**Review by mkannwischer — APPROVED** [review]
LGTM now. Thanks @potsrevennil!

**Pull request merged**